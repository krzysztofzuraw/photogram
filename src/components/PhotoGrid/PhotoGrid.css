.photo-grid {
    display: flex;
    flex-wrap: wrap;
    flex-direction: row;
}

.photo-grid-column {
    padding: 1rem;
}

.photo-grid-header {
    font-family: 'Oswald', sans-serif;
    text-align: center;
    font-weight: 100;
    color: green;
    font-size: 4rem;
    text-shadow: 0px 4px 0 rgba(18, 86, 136, 0.11);
}
