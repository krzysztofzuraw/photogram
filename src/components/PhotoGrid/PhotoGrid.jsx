import React from "react";
import { connect } from "react-redux";

import Photo from "../Photo/Photo.jsx";
import { Grid, Header } from "semantic-ui-react";
import { fetchPhotos } from "../../actions/index.js";

import "./PhotoGrid.css";

class PhotoGrid extends React.Component {
  componentDidMount() {
    this.props.fetchPhotos();
  }

  render() {
    const { photos } = this.props;
    return (
      <div>
        <Header as="h1" textAlign="center" className="photo-grid-header">
          Unsplashgram
        </Header>
        <Grid columns={3} className="photo-grid">
          {photos.map((photo, key) => (
            <Grid.Column className="photo-grid-column" key={key}>
              <Photo />
            </Grid.Column>
          ))}
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  photos: state.photos.photos
});

const mapDispatchToProps = dispatch => ({
  fetchPhotos: () => dispatch(fetchPhotos())
});

export default connect(mapStateToProps, mapDispatchToProps)(PhotoGrid);
