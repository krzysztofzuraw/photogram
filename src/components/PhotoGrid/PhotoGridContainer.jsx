import React from "react";
import { connect } from "react-redux";

import { fetchPhotos } from "../../actions/index.js";

import PhotoGrid from "./PhotoGrid";

class PhotoGridContainer extends React.Component {
  componentDidMount() {
    this.props.fetchPhotos();
  }

  render() {
    return <PhotoGrid photos={this.props.photos} />;
  }
}

const mapStateToProps = state => ({
  photos: state.photos.photos
});

const mapDispatchToProps = dispatch => ({
  fetchPhotos: () => dispatch(fetchPhotos())
});

export default connect(mapStateToProps, mapDispatchToProps)(PhotoGridContainer);
