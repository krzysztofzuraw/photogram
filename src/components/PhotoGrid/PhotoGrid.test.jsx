import React from "react";

import { shallow } from "enzyme";

import PhotoGrid from "./PhotoGrid";

describe("PhotoGrid Component", () => {
  it("renders correctly without photos", () => {
    const wrapper = shallow(<PhotoGrid photos={[]} isFetching={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with photos", () => {
    const photos = [
      {
        id: "1",
        urls: {
          small: "http"
        },
        user: {
          name: "Tom",
          links: {
            self: "https"
          }
        }
      }
    ];
    const wrapper = shallow(<PhotoGrid photos={photos} isFetching={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
