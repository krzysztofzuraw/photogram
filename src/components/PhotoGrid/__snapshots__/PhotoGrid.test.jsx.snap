// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGrid Component renders correctly with photos 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhotoGrid
    isFetching={false}
    photos={
        Array [
            Object {
              "id": "1",
              "urls": Object {
                "small": "http",
              },
              "user": Object {
                "links": Object {
                  "self": "https",
                },
                "name": "Tom",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        Array [
          <div
            className="panel"
>
            <Connect(PhotoContainer)
                        id="1"
                        style={
                                    Object {
                                                "maxWidth": 400,
                                              }
                        }
                        url="http"
                        user={
                                    Object {
                                                "links": Object {
                                                  "self": "https",
                                                },
                                                "name": "Tom",
                                              }
                        }
            />
</div>,
        ],
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <Connect(PhotoContainer)
            id="1"
            style={
                        Object {
                                    "maxWidth": 400,
                                  }
            }
            url="http"
            user={
                        Object {
                                    "links": Object {
                                      "self": "https",
                                    },
                                    "name": "Tom",
                                  }
            }
/>,
          "className": "panel",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "1",
            "style": Object {
              "maxWidth": 400,
            },
            "url": "http",
            "user": Object {
              "links": Object {
                "self": "https",
              },
              "name": "Tom",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          Array [
            <div
              className="panel"
>
              <Connect(PhotoContainer)
                            id="1"
                            style={
                                          Object {
                                                        "maxWidth": 400,
                                                      }
                            }
                            url="http"
                            user={
                                          Object {
                                                        "links": Object {
                                                          "self": "https",
                                                        },
                                                        "name": "Tom",
                                                      }
                            }
              />
</div>,
          ],
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PhotoContainer)
              id="1"
              style={
                            Object {
                                          "maxWidth": 400,
                                        }
              }
              url="http"
              user={
                            Object {
                                          "links": Object {
                                            "self": "https",
                                          },
                                          "name": "Tom",
                                        }
              }
/>,
            "className": "panel",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "1",
              "style": Object {
                "maxWidth": 400,
              },
              "url": "http",
              "user": Object {
                "links": Object {
                  "self": "https",
                },
                "name": "Tom",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`PhotoGrid Component renders correctly without photos 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhotoGrid
    isFetching={false}
    photos={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        Array [],
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          Array [],
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
